src/main/java/com/tech_symfony/movie_booking/api/user/MailController.java

```diff
@@ -0,0 +1,79 @@
+package com.tech_symfony.movie_booking.api.user;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeRequestUrl;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeTokenRequest;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.io.IOException;
+import java.util.Collections;
+
+//Class này dùng để lấy refresh token từ gg để dùng Gmail API
+//Trước tiên phải enable Gmail API trong gg dev console, email dùng để gửi thư phải là account của gg dev console đó
+//gọi /api/v1/auth/mail trong trình duyệt để tiến hành đăng nhập và app sẽ lấy refresh token để dùng
+//chi tiết config trong class MailConfig
+//lưu ý redirect uri phải được đăng ký đúng (trong trường hợp đang sử dụng là http://localhost:8080/api/v1/auth/callback)
+@Controller
+@RequestMapping("/api/v1")
+@RequiredArgsConstructor
+public class MailController {
+
```

Comment:
@RepositoryRestController
@ResponseBody
Dòng 25 26 thay 2 dòng trên nha vì domain chính thức của mình là api.techsymphony.io.vn 
còn v1 thì mình dùng thư viện spring data rest nên   
@RepositoryRestController
dòng này sẽ giúp mình config toàn bộ api của các phần khác chung luôn thành v0, v1, v2 ,....